src/Http/Owin/src/OwinEnvironment.cs

```diff
@@ -478,4 +400,229 @@ public FeatureMap(Func<TFeature, object> getter, Func<object> defaultFactory, Ac
+            };
+        }
+
+        static string PortToString(int port) => port switch
+        {
+            80 => "80",
+            443 => "443",
+            _ => port.ToString(CultureInfo.InvariantCulture),
+        };
+
+        public int Count
+        {
+            get
+            {
+                if (_contextEntries is not null)
+                {
+                    return _contextEntries.Count;
+                }
+
+                return _entries.Count + _contextDependentEntries.Count - (_deletedKeys?.Count ?? 0);
+            }
+        }
+
+        public IDictionary<string, FeatureMap> GetFeatureMaps()
+        {
+            InitializeContextEntries();
+            return _contextEntries;
+        }
+
+        public bool TryGetValue(string key, out FeatureMap entry)
+        {
+            if (_contextEntries is not null)
+            {
+                return _contextEntries.TryGetValue(key, out entry);
+            }
+
+            if (_deletedKeys?.Contains(key) == true)
+            {
+                entry = null;
+                return false;
+            }
+
+            if (_entries.TryGetValue(key, out entry))
+            {
+                return true;
+            }
+
+            return _contextDependentEntries.TryGetValue(key, out entry);
+        }
+
+        public bool ContainsKey(string key)
+        {
+            if (_contextEntries is not null)
+            {
+                return _contextEntries.ContainsKey(key);
+            }
+
+            if (_deletedKeys?.Contains(key) == true)
+            {
+                return false;
+            }
+
+            return _entries.ContainsKey(key) || _contextDependentEntries.ContainsKey(key);
+        }
+
+        public bool Remove(string key)
+        {
+            if (_contextEntries is not null)
+            {
+                return _contextEntries.Remove(key);
+            }
+
+            if (_entries.ContainsKey(key) || _contextDependentEntries.ContainsKey(key))
+            {
+                _deletedKeys ??= new HashSet<string>();
+                return _deletedKeys.Add(key);
+            }
+
+            return false;
+        }
+
+        public IEnumerator<KeyValuePair<string, FeatureMap>> GetEnumerator()
+        {
+            if (_contextEntries is not null)
+            {
+                foreach (var entry in _contextEntries)
+                {
+                    yield return entry;
+                }
+            }
+            else
+            {
+                foreach (var entry in _entries)
+                {
+                    if (_deletedKeys?.Contains(entry.Key) == true)
+                    {
+                        continue;
+                    }
+
+                    yield return entry;
+                }
+
+                foreach (var entry in _contextDependentEntries)
+                {
+                    if (_deletedKeys?.Contains(entry.Key) == true)
```

Comment:
```suggestion
                    if (_deletedKeys?.Contains(entry.Key))
```
?

Reply:
same elsewhere

Reply:
every place `== true` is being used there is a check on nullability as well: `if (_deletedKeys?.<some check> == true)`. Removing `== true` gives the build time error:
![image](https://github.com/user-attachments/assets/52b0b1c9-f6c7-423c-a73c-5f200d6657a2)

am I missing anything?

Reply:
No, I was reading too fast